### PR TITLE
Nix: Expose .env feature flags to Nix as attrset

### PR DESCRIPTION
**Pull request by pedropombeiro**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

### Summary

[comment]: # (Summarise the problem and how the pull request solves it)
This PR allows Nix code to read feature flags present in the correct .env file.

Currently only tested for Android builds (since this is where we'll use this for Nimbus).

status: ready

### CODE REVIEW

File: nix/env.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# This Nix expression takes care of reading/parsing the correct .env configuration file and return it as an attr set
```
Comment (jakubgs) [inline nix/env.nix]: Move this to `nix/tools/envParser.nix`.

File: default.nix [hunk 2]
```diff
@@ -1,9 +1,10 @@
 {
   config ? { }, # for passing status_go.src_override
+  build-type ? "nightly",
 }:
 
 let
-  main = import ./nix/default.nix { inherit config; };
+  main = import ./nix/default.nix { inherit config build-type; };
```
Comment (jakubgs) [inline default.nix]: Please move `build-type` into `config`. No need to make it special.

**Comment by jakubgs** [comment]
If changing contents of `config` causes unnecessary rebuilds lets just ditch it and put our own stuff into `params`, including the `status-go` path override thing. It should be passed around implicitly by `callPackage` if it is required as an argument by called Nix expression, so the code should remain clean except for cases when we call `import` directly.

**Comment by pedropombeiro** [comment]
I did some tests and changing values inside config doesn't cause unnecessary rebuilds. I'll redo this PR using `config` sub-values.

**Review by jakubgs — APPROVED** [review]
LGTM. But maybe you could add some comments explaining our use of `config`. And possible a section in `nix/README.md` explaining that in more detail.

**Review by jakubgs — APPROVED** [review]
Very nice, thanks for the README update.

**Pull request merged**